src/org/rascalmpl/uri/file/MavenRepositoryURIResolver.java

```diff
@@ -0,0 +1,190 @@
+package org.rascalmpl.uri.file;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.rascalmpl.uri.URIUtil;
+import org.rascalmpl.uri.jar.JarURIResolver;
+
+import io.usethesource.vallang.ISourceLocation;
+
+/**
+ * Finds jar files (and what's inside) relative to the root of the LOCAL Maven repository.
+ * 
+ * We use `mvn://<groupid>.<name>-<version>/<path-inside-jar>` as the general scheme.
+ * So the authority encodes the identity of the maven project and the path encodes
+ * what's inside the respective jar file.
+ * 
+ * Here version is [0-9]+ major . [0-9]+ minor .[0-9]+ path -[A-Za-z\-]* optionalTag
+ * 
+ * Locations with the `mvn` scheme are typically produced by configuration code that uses 
+ * Maven to resolve dependencies. Once the group id, name and version are known, any
+ * `mvn:///` location is easily constructed and uses. It can be seen as a transparent
+ * short-hand for an absolute `file:///` location that points into the (deeply nested)
+ * .m2 local repository. The prime benefits are:
+ *    1. much shorter location than `file:///`
+ *    2. full transparency, more so than `lib:///`
+ *    3. unique identification, modulo the (configured) location of the local repository.
+ * 
+ * It is a logical resolver in order to allow for short names for frequently
+ * occurring paths, without loss of transparancy. We always know which jar file is meant,
+ * and the encoding is one-to-one. 
+ * 
+ * This resolver does NOT find the "latest" version or any version of a package without an explicit
+ * version reference in the authority pattern, by design. Any automated resolution here would
+ * make the dependency management downstream less transparant and possibly error-prone. 
+ * 
+ * This resolver also does not implement any download or installation procedure, by design.
+ * It should simply reflect what has been downloaded and installed into the LOCAL maven
+ * repository.
+ * 
+ * This resolver is to replace for the large part the use of the deprecate `lib` scheme
+ * from {@see RascalLibraryURIResolver} which leaves too much implicit and automates too 
+ * much to obtain any transparancy in dependency resolution. 
+ */
+public class MavenRepositoryURIResolver extends AliasedFileResolver {
+    private final Pattern authorityRegEx 
+        = Pattern.compile("^([a-zA-Z0-9-_.]+?)[!]([a-zA-Z0-9-_.]+)([!][a-zA-Z0-9\\-_.]+)$");
+    //                               groupId         !  artifactId      ! optionAlComplexVersionString
```

Comment:
/ is not allowed in authorities

Reply:
We use ! here too to go into the jar. See below.

Reply:
This is why the ! is a good character to use. We've used it before to avoid overlap with filenames; and that seemed ok. Every other character that is not a / or a ! or a . is a bad separator because people are allowed to use then in filenames.

Reply:
. ended up not to work because people don't follow the maven naming rules.

Reply:
I meant, groupid as authority, and the rest for regular paths? 

Alternative: we could even use authority to distinguish between local and remote maven repo

Reply:
* Remote maven repo's are something to think about very carefully, in a legal sense. I don't want to blur that distinction anywhere. So for this scheme that is out of the question.
* I want the authority to identify a jar, a project scope, just like the other schemes we use for this, project, plugin, bundle, etc. That way the path is always consistently rooted at the base of a file name space. I don't want our Rascal code to have to split loc paths based on some arbitrary knowledge about the specific scheme.

Reply:
The last point makes code that runs directly from target folders or project schemes _and_ from an mvn jar, or from an eclipse plugin, or OSGI bundle, etc. etc. much easier to make correct, because `loc.path` will produce the same postfix in every situation.